guides/git-pr-review.md

```diff
@@ -0,0 +1,102 @@
+# GitHub Pull Request Review ("Code Review")
+
+### Introduction
+A code review for a GitHub Pull Request (PR) could mean many things but likely
+it should consist of a few steps that are beyond just "reviewing the code".
+Below are some steps that could be used as a guide in the review of a PR.
+Remember, too, that a review should respectful and that we have 
+a [Code of Conduct](https://github.com/spacetelescope/style-guides/blob/master/CODE_OF_CONDUCT.md) 
+that we follow.
+
+### Who Should Review
+There are several groups of people who might be interested in reviewing a PR.
+Members of the development team, package maintainer (GitHub repository
+maintainer), or scientists / product owners.  Each group may have a different
+focus on the steps below, for example:
+
+* The development team might focus more on Steps 2 and 3 (obviously with
+  knowledge of Step 1).
+
+* A scientist, content specialist, or product owner might focus more on Steps 4
+  and 5, so they can confirm that the changes make sense from a science
+  point-of-view. 
+
+* The package maintainer / GitHub repository maintainer will want to make sure
+  that each of these steps are done but may not need to be the one doing each
+  step.  The package maintainer will be the one who will want to confirm the code
+  to be merged fits the framework of existing code, is correct and the
+  documentation and tests are sufficient. 
+
+
+### PR Review Steps
+
+#### 1. Review the PR and Issue
+
+Read through the PR to see what was done and the discussion people have had on
+there. Confirm the GitHub "fixes" or "closes" magic words are used to
+automatically close the related issues.
+
+Then, if there is a corresponding issue or issues, read through the original
+content of the issue and then read through the discussion and clarification.
+
+Many times there are going to be other "requirements" or clarifications that 
+might influence the PR beyond the original issue text.
+
+If the PR does not appear to fix/implement the issue or the description is not
+clear in the PR text, then stop here and ask questions. Document any new
+understanding or thoughts in the PR for future reference.
+
+#### 2. Review the code
```

Comment:
Somewhere in this section you probably want to link to the Python code-specific style guide (currently that's just Python - https://github.com/spacetelescope/style-guides/blob/master/guides/python.md ), right?  I.e., "check that it matches the ST style recommendations"?

(More review later, but this popped up at a first glance)

Reply:
I added some text to that point.